File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.cpp

class Solution {
public:
    void setZeroes(vector<vector<int>>& matrix) {
        vector<vector<int>> ans=matrix;
        for(int i=0;i<matrix.size();i++){
            for(int j=0;j<matrix[i].size();j++){
                if(matrix[i][j] == 0){
                    for(int k=0;k<matrix.size();k++)
                        ans[k][j]=0;
                    for(int k=0;k<matrix[i].size();k++)
                        ans[i][k]=0;
                }
            }
        }
        matrix=ans;
    }
};

File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.java
class Solution {
    public void setZeroes(int[][] matrix) {
        int m=matrix.length;
        int n=matrix[0].length;
        int horArr[]=new int[n];
        int verArr[]=new int[m];
        Arrays.fill(horArr,0);
        Arrays.fill(verArr,0);
        
        for(int i=0;i<m;i++){
            for(int j=0;j<n;j++){
                if(matrix[i][j] == 0){
                    horArr[j]=1;
                    verArr[i]=1;
                }
            }
        }
        for(int i=0;i<m;i++){
            for(int j=0;j<n;j++){
                if(horArr[j] == 1 || verArr[i] == 1){
                    matrix[i][j]=0;
                }
            }
        }
    }
}